app/api/submission/package.scala

```diff
@@ -62,32 +62,49 @@ package object submission {
         case value: JsArray => JsSuccess(value.readValuesAs[T])
         case _              => throw new Exception(s"$path did not contain an array")
       }
+
+    def readFilteredArray[T](f: JsValue => Boolean)(implicit reads: Int => Reads[T]): Reads[Seq[T]] =
+      path.readWithDefault[Seq[T]](Nil) {
+        case value: JsArray => JsSuccess(value.readFilteredValuesAs[T](f))
+        case _              => throw new Exception(s"$path did not contain an array")
+      }
```

Comment:
Really like this. Reminiscent of the UR readers on the frontend. A recommendation would be to maybe have an [Option[NonEmptyList]] to help identify if its empty or not (though this is just a personal preference)

Reply:
It's an interesting one. Part of me wants the reads to be as 'dumb' as possible i.e. we work on the assumption that our navigation/clean-up in the frontend means that what gets delivered to the backend is a valid set of user answers. All the backend does is map that to a different shape.